### PR TITLE
Attempt to automate release

### PR DESCRIPTION
**Pull request by roxyboy**
I started working on the process to automate package distribution releases (following [this](https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/)) but... I'm new to this so any feedback would be much appreciated.

### CODE REVIEW

**Comment by rabernat** [comment]
I recommend copying what we do in xgcm:

https://github.com/xgcm/xgcm/blob/master/.github/workflows/pythonpublish.yaml

**Comment by roxyboy** [comment]
@rabernat I copied the xgcm yaml file. Does this now mean that if I merge this and release a new version, it will automatically be pushed to pypi?

File: .github/workflows/pythonpublish.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        with:
+          python-version: '3.x'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install setuptools setuptools-scm wheel twine
+      - name: Build and publish
+        env:
+          TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
```
Comment (rabernat) [inline .github/workflows/pythonpublish.yaml:23]: Have you set the secrets on this repo?

Comment (rabernat) [reply .github/workflows/pythonpublish.yaml:23]: We should be able to get the secrets at the organization level.

**Comment by rabernat** [comment]
Let's merge this and see if it works!

**Pull request merged**